src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/QCallHandles.cs

```diff
@@ -10,7 +10,7 @@ namespace System.Runtime.CompilerServices
     // Wrapper for address of a string variable on stack
     internal unsafe ref struct StringHandleOnStack
     {
-        private void* _ptr;
+        private readonly void* _ptr;
```

Comment:
This field can be changed from the unmanaged code in the runtime. Do we need to worry about the JIT optimizing around the fact that this can't be changed in C# with the `readonly` modifier? (Applies to all types in this file).

Reply:
Not currently, but I also intended to avoid changing such cases where the data was mutated in native, so I'll undo these.